src/main/resources/templates/db_scripts/migration/extract_credits_from_expenditures.sql

```diff
@@ -4,20 +4,20 @@
 -- 3. Subtract from expenditures amount
 
 WITH credit_sums AS (
-    SELECT
-        budget.id AS budget_id,
-        COALESCE(SUM((trx.jsonb->>'amount')::numeric), 0) AS total_credits
-    FROM ${myuniversity}_${mymodule}.budget AS budget
-    LEFT JOIN ${myuniversity}_${mymodule}.transaction AS trx
-        ON trx.fiscalyearid = budget.fiscalyearid AND trx.tofundid = budget.fundid
-        AND trx.jsonb->>'transactionType' = 'Credit'
-    GROUP BY budget.id
+   SELECT
+      budget.id AS budget_id,
+      SUM(CASE WHEN trx.jsonb->>'transactionType' = 'Credit' THEN (trx.jsonb->>'amount')::numeric ELSE 0 END) AS total_credits,
+      SUM(CASE WHEN trx.jsonb->>'transactionType' = 'Payment' THEN (trx.jsonb->>'amount')::numeric ELSE 0 END) AS total_expenditures
+   FROM ${myuniversity}_${mymodule}.budget AS budget
```

Comment:
This switch of tables is a good, the budget is always there but its associated transactions are optional